src/report/ReportPoll.ts

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Mjolnir, REPORT_POLL_EVENT_TYPE } from "../Mjolnir";
+import { ReportManager } from './ReportManager';
+import { LogLevel } from "matrix-bot-sdk";
+
+class InvalidStateError extends Error {}
+
+export class ReportPoll {
+    private from = 0;
+    private timeout: ReturnType<typeof setTimeout> | null = null;
+
+    /*
+     * A class to poll synapse's report endpoint, so we can act on new reports
+     *
+     * @param client The Matrix client underpinning the running Mjolnir
+     * @param manager The report manager in to which we feed new reports
+     * @param save An abstract function to persist where we got to in report reading
+     */
+    constructor(
+        private mjolnir: Mjolnir,
+        private manager: ReportManager,
+    ) { }
+
+    private schedulePoll() {
+        const self = this;
+        this.timeout = setTimeout(
+            function() { self.getAbuseReports() },
```

Comment:
>is using something like bind more typescriptic?

(I don't know why GitHub is not letting me reply to the original comment but regardless)

it's probably more precise to use bind. After having a look just to double check whether I'm being silly, bind has got this exact use case as an example on MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind#with_settimeout